erpnext/controllers/queries.py

```diff
@@ -216,11 +216,22 @@ def item_query(doctype, txt, searchfield, start, page_len, filters, as_dict=Fals
 		if not field in searchfields]
 	searchfields = " or ".join([field + " like %(txt)s" for field in searchfields])
 
+	if filters.get('supplier'):
```

Comment:
@noahjacob, @rohitwaghchaure filters argument is a string containing json, when called from the frontend. The following code breaks with this PR:

```js
frappe.call({
    method: 'erpnext.controllers.queries.item_query',
    args: {
        doctype: 'Item',
        txt: '',
        searchfield: 'name',
        start: 0,
        page_len: 99,
        filters: {
            supplier: 'Unico Plastics Inc.',
            is_sales_item: 1
        },
        as_dict: true
    },
    callback: (r) => console.log(r.message)
});
```

filters string needs to be parsed to json first.

Please fix or revert this PR.

Reply:
already fixed via #25915 , will release in sometime

Reply:
@nabinhait while it doesn't produce an error anymore, filters will still be ignored in my example.